src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java

```diff
@@ -19,6 +22,17 @@ public CustomerService(CustomerRepository customerRepository) {
         this.customerRepository = customerRepository;
     }
 
+    public Customer findByUsername(String username) throws BadRequestException {
+
+        Optional<Customer> optionalCustomer = this.customerRepository.findById(username);
+
+        if (optionalCustomer.isPresent()) {
+            return optionalCustomer.get();
+        } else {
+            throw new BadRequestException("username does not exist");
```

Comment:
should be a 404 when you cant find the resource queried, I think.

Reply:
👍